src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -134,9 +143,15 @@ class PeerManagerActor(
   }
 
   private def maybeConnectToDiscoveredNodes(connectedPeers: ConnectedPeers, nodes: Set[Node]): Unit = {
-    val nodesToConnect = nodes
+    val discoveredNodes = nodes
       .filter(connectedPeers.canConnectTo)
-      .take(connectedPeers.outgoingConnectionDemand)
+
+    val nodesToConnect = discoveredNodes
+      .filterNot(n => triedNodes.contains(n.id)) match {
+      case seq if seq.size > connectedPeers.outgoingConnectionDemand =>
```

Comment:
Can the `if` can be `>=` instead?

Reply:
no, because seq is a set of nodes, and if condition is on it's size

Reply:
or you meant not removing the if but changing the condition? :) yes, it makes sense

Reply:
Haha sorry, I will be more verbose next time :)